### PR TITLE
Add BeforeTransport policy position

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

**Comment by KrzysztofCwalina** [comment]
What behavior do we want when more than 1 BeforeTransport policy is added?

**Comment by pakrym** [comment]
>What behavior do we want when more than 1 BeforeTransport policy is added?

Same as with other positions, they would be added in the same order they were added to client options.

**Review by KrzysztofCwalina — APPROVED** [review]

File: sdk/core/Azure.Core/src/ClientOptions.cs [hunk 1]
```diff
@@ -86,22 +86,18 @@ public HttpPipelineTransport Transport
...
             }
-        }
 
-        internal IList<HttpPipelinePolicy> PerCallPolicies { get; }
+            Policies ??= new();
+            Policies.Add((position, policy));
+        }
 
-        internal IList<HttpPipelinePolicy> PerRetryPolicies { get; }
+        internal List<(HttpPipelinePosition Position, HttpPipelinePolicy Policy)>? Policies { get; private set; }
```
Comment (tg-msft) [inline sdk/core/Azure.Core/src/ClientOptions.cs:100]: Are there any size issues with using tuples?  KVP might be cheaper?

**Review by tg-msft — APPROVED** [review]
Looks good

File: sdk/core/Azure.Core/src/ClientOptions.cs (hunk 1 above)
Comment (pakrym) [reply sdk/core/Azure.Core/src/ClientOptions.cs:100]: No issues AFAIK. Both are defined with 2 properties/fields.

Comment (tg-msft) [reply sdk/core/Azure.Core/src/ClientOptions.cs:100]: Yeah, sorry, these get turned into `ValueTuple<...>` which means no large ToString, etc.

**Pull request merged**